### PR TITLE
Add GoPhish helper scripts

### PR DESCRIPTION
**Pull request by dav3r**
# 

## 🗣 Description
This PR adds four helper scripts make it easier to call various scripts in the [`gophish-tools`](https://github.com/cisagov/gophish-tools) Docker container:
 
- `import_assessment.sh`: Simplifies the process of importing an assessment JSON file into the GoPhish server running in the local Docker composition (calls `gophish-tools/gophish-import` and `gophish-tools/gophish-complete`).
- `export_assessment.sh`: Simplifies the process of exporting assessment data from the GoPhish server running in the local Docker composition to a JSON file (calls `gophish-tools/gophish-export`).
- `complete_campaign.sh`: Simplifies the process of completing a campaign on the GoPhish server running in the local Docker composition (calls `gophish-tools/gophish-complete`).
- `test_assessment.sh`: Simplifies the process of sending test emails for an assessment in the GoPhish server running in the local Docker composition (calls `gophish-tools/gophish-test`).

## 💭 Motivation and Context
These helper scripts are intended to simplify the day-to-day workflow of and minimize errors by Phishing Campaign Assessment (PCA) operators.

## 🧪 Testing
All of these scripts were successfully tested in COOL `env1-staging`.

## 🚥 Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (causes existing functionality to change)

## ✅ Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/scripts/import_assessment.sh [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+then
+  echo "Assessment successfully imported from $ASSESSMENT_FILE!"
+  echo ""
+else
+  echo "ERROR: Assessment import from $ASSESSMENT_FILE failed!"
+  exit $import_rc
+fi
+
+# TODO - REMOVE THIS TEMP HACK
+ASSESSMENT_FILE="$1.json"
```
Comment (jsf9k) [inline src/scripts/import_assessment.sh:50]: This is pretty ugly.  Why not use two variables?  Call the other variable `ASSESSMENT_ID` or something similar and create `ASSESSMENT_FILE="${ASSESSMENT_ID}.json"` from it.  (Use whatever names make sense.)

File: src/scripts/complete_campaign.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+# complete_campaign.sh CAMPAIGN_ID
+
+# This script simplifies the process of completing a campaign on the
+# GoPhish server running in the local Docker composition.
+
+set -o nounset
```
[the remark this answers is not in the document]
Comment (dav3r) [reply src/scripts/complete_campaign.sh:8]: When `errexit` is set, the script exits before I can capture any error code, which means that I can't output those helpful error messages.  I'm open to alternatives here.

[the remark this answers is not in the document]
Comment (jsf9k) [reply src/scripts/complete_campaign.sh:8]: In the past I have turned on `errexit` at the top of a script, but then turned it off (`set +o errexit`) just before I run the command that errors out.  Then I turn it back on right afterwards.  See, for example, [here](https://github.com/cisagov/ansible-role-freeipa-server/blob/develop/templates/disable_inactive_users.j2.sh#L85-L96).

[the remark this answers is not in the document]
Comment (dav3r) [reply src/scripts/complete_campaign.sh:8]: That will work- thanks!

File: src/scripts/import_assessment.sh (hunk 1 above)
Comment (dav3r) [reply src/scripts/import_assessment.sh]: This was just a temporary thang until @bjb28 updated one of the scripts in [`gophish-tools`](https://github.com/cisagov/gophish-tools), which he did [here](https://github.com/cisagov/gophish-tools/commit/5e2dc21614d2e6a7a8d313b510ad316a18de6a04#diff-732e89c3e3a7866d8c19dd0bfe0d38a9L276).  I just forgot to undo my hack before I put in this PR.  That has been rectified with https://github.com/cisagov/pca-gophish-composition-packer/pull/13/commits/b799060c2a764a7fd5f231bab4d2e5172e08fc8a.

File: src/scripts/complete_campaign.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (dav3r) [reply src/scripts/complete_campaign.sh:9]: `errexit` goodness added in https://github.com/cisagov/pca-gophish-composition-packer/pull/13/commits/f004e05d480b4edfae65d57c4a8086360e42c83f.

File: src/scripts/export_assessment.sh [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+if [ $# -ne 1 ] || [ "$1" == "-h" ] || [ "$1" == "--help" ]
+then
+    echo "Usage: export_assessment.sh ASSESSMENT_ID"
+    exit 255
+fi
+
+ASSESSMENT_ID=$1
+
+GOPHISH_COMPOSITION="/var/pca/pca-gophish-composition/docker-compose.yml"
```
[the remark this answers is not in the document]
Comment (dav3r) [reply src/scripts/export_assessment.sh]: This is an excellent suggestion, which I implemented in https://github.com/cisagov/pca-gophish-composition-packer/pull/13/commits/bb26346345192831597a98b1b1a37a35edb7b1e0 and https://github.com/cisagov/pca-gophish-composition-packer/pull/13/commits/83143f75be139018c7e899271575a8181b3572ec.

Note that I had to add an extra argument for the shellcheck pre-commit hook - see https://github.com/cisagov/pca-gophish-composition-packer/pull/13/commits/ba25f83ef74da9506f9bba47a8fe97fbd88a4112 for details.

File: src/scripts/import_assessment.sh [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+# Schedule each campaign to be completed at the specified time
+# via the "at" command
+for campaign in $(jq '.campaigns | keys | .[]' "$ASSESSMENT_FILE"); do
```
[the remark this answers is not in the document]
Comment (dav3r) [reply src/scripts/import_assessment.sh]: 👍 See https://github.com/cisagov/pca-gophish-composition-packer/pull/13/commits/d242789382735fd43f68843426ff7bb73528b645.

**Comment by jsf9k** [comment]
Nice work!  I think this is looking really good.

**Comment by dav3r** [comment]
@jsf9k and @hillaryj - I re-requested your review to highlight a small, but important step that I left out previously- please see https://github.com/cisagov/pca-gophish-composition-packer/pull/13/commits/538c36275a411c9f1a1d13a12a98ba8f6460dc1e.

**Comment by dav3r** [comment]
> Looking good. I do have a 🔧 to throw in this whole thang though.
> 
> These scripts seem generally useful even if you're not using an AMI. I think these might fit better in the
> [cisagov/pca-gophish-composition](https://github.com/cisagov/pca-gophish-composition) repository.
> 
> It puts the code closer to the implementation of the composition so it should easier to test as the composition evolves. That is, these scripts are more tightly-coupled to the composition than the packer build, and should live within the same testing domain.
> 
> Also, we have a similar "helper" script already in that repository:
> https://github.com/cisagov/pca-gophish-composition/tree/develop/src/gophish_init
> 
> What do you think?

@felddy I think you are correct.  When I was discussing this originally with @jsf9k, I think he was trying to make a similar point, but I wasn't convinced at the time- now I am sold.  

I will go ahead and move these scripts to `pca-gophish-composition`.  Are you ok with me putting them in a `pca-gophish-composition/gophish-tools`directory, similar to what I was doing here: https://github.com/cisagov/pca-gophish-composition-packer/pull/13/commits/538c36275a411c9f1a1d13a12a98ba8f6460dc1e?  From a PCA operator standpoint, I think it will be simpler for them to find them there vs. if they are tucked away in `src/`.

**Comment by dav3r** [comment]
At @felddy's suggestion, these scripts are better-suited to living in https://github.com/cisagov/pca-gophish-composition, so I created https://github.com/cisagov/pca-gophish-composition/pull/15 and I am closing this PR.

**Pull request closed without merging**